src/explore-education-statistics-frontend/src/modules/api/permalink/__tests__/createPermalinkTable.test.ts

```diff
@@ -0,0 +1,238 @@
+import createPermalinkTable from '@frontend/modules/api/permalink/createPermalinkTable';
+import {
+  createApiMocks,
+  createApiRequest,
+  createApiResponse,
+} from '@frontend-test/apiRouteMocks';
+import type { NextApiRequest, NextApiResponse } from 'next';
+
+describe('permalink api route', () => {
+  type ApiRequest = NextApiRequest & ReturnType<typeof createApiRequest>;
+  type ApiResponse = NextApiResponse & ReturnType<typeof createApiResponse>;
+
+  describe('authorised', () => {
+    test('returns a 400 when there is no body on the request', async () => {
+      const { req, res } = createApiMocks({
+        method: 'POST',
+      });
+
+      await createPermalinkTable(req, res);
+
+      expect(res.statusCode).toBe(400);
+      // eslint-disable-next-line no-underscore-dangle
+      expect(res._getData()).toEqual({
+        message: 'fullTable and configuration required',
+        status: 400,
+      });
+    });
+
+    test('returns a 400 when the body does not contain fullTable and configuration', async () => {
+      const {
+        req,
+        res,
+      }: { req: ApiRequest; res: ApiResponse } = createApiMocks({
```

Comment:
We don't need the explicit type anymore if we use `createApiMocks`. Same applies to other places in this test suite